gridded_etl_tools/dataset_manager.py

```diff
@@ -95,9 +92,10 @@ def __init__(
         skip_pre_parse_nan_check: bool = False,
         skip_post_parse_qc: bool = False,
         skip_post_parse_api_check: bool = False,
-        encryption_key: str = None,
+        encryption_key: str | None = None,
         use_compression: bool = True,
         dry_run: bool = False,
+        output_zarr3: bool = False,
```

Comment:
I think we should  very aggressively be checking if the user is setting `output_zarr3` on an existing v2 zarr, or vice versa. I don't fully trust `zarr` to raise an appropriate exception and not brick our dataset, as it's had issues with that in the past